sass/woocommerce/_archives.scss

```diff
@@ -83,7 +83,7 @@
 				text-align: center;
 
 				@media (max-width: 768px) {
-					& {
+					@at-root body.responsive#{&} {
```

Comment:
This compiles to `body.responsive.woocommerce`. It is getting compiled without any error on my system. Not sure when it was introduced, but guess an update to your SASS compiler should do the trick.